complete_streaming_function.py

```
@@ -0,0 +1,394 @@
+                    ConversationLog.user_progress_id == user_progress.id
+                ).order_by(desc(ConversationLog.message_order)).first()
+                begin_order = (last_msg.message_order + 1) if last_msg else 1
+                
+                begin_user_log = ConversationLog(
+                    user_progress_id=user_progress.id,
+                    scene_id=user_progress.current_scene_id,
+                    message_type="user",
+                    sender_name="User",
+                    message_content=request.message,
+                    message_order=begin_order,
+                    timestamp=datetime.utcnow()
```

Ruff (0.14.1)

Undefined name `datetime`

(F821)